src/Compilers/CSharp/Portable/Binder/Semantics/Conversions/ConversionsBase.cs

```diff
@@ -2731,6 +2742,18 @@ private bool HasVariantConversionNoCycleCheck(NamedTypeSymbol source, NamedTypeS
                 destinationTypeArguments.Free();
             }
 
+            static bool isTypeIEquatable(NamedTypeSymbol type)
+            {
+                return type is
```

Comment:
What does this allow that `type.Equals(compilation.GetWellKnownType(System_IEquatable_T))` does not?

Reply:
I see the comment above about "the canonical IEquatable", just not certain I know what that means except to guess that somehow the compiler decides that one of the IEquatable definitions is more "official" than the others?

Reply:
There's always a single canonical well-known type, so if you do something like define IEquatable in your compilation, then reference another assembly that defines or has another definition of IEquatable, then that one will not be considered as being well-known.

In contrast, we have things like tuples, where *every* definition of System.ValueTuple we find is considered a tuple type.

I think this should be in the latter category, because we may be trying to find out whether a conversion exists between two metadata types and have our own well-known IEquatable, and we don't want that conversion to fail.

Reply:
Thanks, makes sense.